src/ProjectService.Validation/EditTaskValidator.cs

```diff
@@ -0,0 +1,138 @@
+        public static string Description => $"/{nameof(EditTaskRequest.Description)}";
+        public static string AssignedTo => $"/{nameof(EditTaskRequest.AssignedTo)}";
+        public static string PriorityId => $"/{nameof(EditTaskRequest.PriorityId)}";
+        public static string StatusId => $"/{nameof(EditTaskRequest.StatusId)}";
+        public static string TypeId => $"/{nameof(EditTaskRequest.TypeId)}";
+        public static string PlannedMinutes => $"/{nameof(EditTaskRequest.PlannedMinutes)}";
+
+        Func<JsonPatchDocument<EditTaskRequest>, string, Operation> GetOperationByPath =>
+            (x, path) => x.Operations.FirstOrDefault(x => x.path == path);
+
+        public EditTaskValidator(
+            ITaskPropertyRepository taskPropertyRepository,
+            IUserRepository userRepository)
+        {
+            _taskPropertyRepository = taskPropertyRepository;
+            _userRepository = userRepository;
+
+            CascadeMode = CascadeMode.Stop;
+
+            RuleFor(x => x.Operations)
+                .Must(x =>
+                    x.Select(x => x.path)
+                        .Distinct().Count() == x.Count())
+                .WithMessage("You don't have to change the same field of Task multiple times.")
+                .Must(x => x.Any())
+                .WithMessage("You don't have changes.")
+                .ForEach(y => y
+                    .Must(x => Paths.Any(cur => string.Equals(
+                        cur,
+                        x.path,
+                        StringComparison.OrdinalIgnoreCase)))
+                    .WithMessage(
+                        $"Document contains invalid path. Only such paths are allowed: {Paths.Aggregate((x, y) => x + ", " + y)}")
+                )
+                .DependentRules(() =>
+                {
+                    When(x => GetOperationByPath(x, Name) != null, () =>
+                    {
+                        RuleFor(x => x.Operations)
+                            .UniqueOperationWithAllowedOp(Name, "replace", "add");
+
+                        RuleFor(x => (string) GetOperationByPath(x, Name).value)
+                            .NotEmpty()
+                            .MaximumLength(150)
+                            .WithMessage("Name is too long");
+                    });
+
+                    When(x => GetOperationByPath(x, Description) != null, () =>
+                    {
+                        RuleFor(x => x.Operations)
+                            .UniqueOperationWithAllowedOp(Description, "replace", "add", "remove");
+
+                        RuleFor(x => GetOperationByPath(x, Description).value)
+                            .NotNull();
+                    });
+
+                    When(x => GetOperationByPath(x, AssignedTo) != null, () =>
+                    {
+                        RuleFor(x => x.Operations)
+                            .UniqueOperationWithAllowedOp(AssignedTo, "replace", "add", "remove");
```

Comment:
why use add, remove?

Reply:
because it can be wasn't assigned and we can remove assign